### PR TITLE
YanZhylavy

### PR DESCRIPTION
**Pull request by YanZhylavy**

### CODE REVIEW

File: YanZhylavy/HW1/Practical task 1.1.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name = input("What is your name?: ")
```
Comment (kolyasalubov) [inline YanZhylavy/HW1/Practical task 1.1.py:1]: good job

File: YanZhylavy/HW1/Practical task 1.2.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name = input("Enter your name, please: ")
```
Comment (kolyasalubov) [inline YanZhylavy/HW1/Practical task 1.2.py:1]: good job

File: YanZhylavy/HW3/Practical task 3.1.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+11.Unless explicitly silenced.
+12.In the face of ambiguity, refuse the temptation to guess.
+13.There should be one-- and preferably only one --obvious way to do it.
+14.Although that way may not be obvious at first unless you're Dutch.
+15.Now is better than never.
+16.Although never is often better than *right* now.
+17.If the implementation is hard to explain, it's a bad idea.
+18.If the implementation is easy to explain, it may be a good idea.
+19.Namespaces are one honking great idea -- let's do more of those!'''
+
```
Comment (kolyasalubov) [inline YanZhylavy/HW3/Practical task 3.1.py:22]: good job

File: YanZhylavy/HW3/Practical task 3.2.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+a = list(input("Please, enter a 4-digit number:"))
```
Comment (kolyasalubov) [inline YanZhylavy/HW3/Practical task 3.2.py:1]: good job

File: YanZhylavy/HW3/Practical task 3.3.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+a = input('Enter a: ')
```
Comment (kolyasalubov) [inline YanZhylavy/HW3/Practical task 3.3.py:1]: good job

File: YanZhylavy/HW4/Practical task 4.1.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+celcius = input('Please enter the temperature value in Celsius to convert it to Fahrenheit: ')
```
Comment (kolyasalubov) [inline YanZhylavy/HW4/Practical task 4.1.py:1]: good job

File: YanZhylavy/HW5/Practical task 5.1.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+list = [34,5,26,78,45,74,8096]
```
Comment (kolyasalubov) [inline YanZhylavy/HW5/Practical task 5.1.py:1]: good job

File: YanZhylavy/HW5/Practical task 5.1.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+for i in list:
```
Comment (kolyasalubov) [inline YanZhylavy/HW5/Practical task 5.1.py]: please look at this implementation:
list_of_number = [2, 1, 4, 6, 1, 4, 8, 6]

for i in range(len(list_of_number)):
    list_of_number[i] = float(list_of_number[i])
print(list_of_number)

File: YanZhylavy/HW6/Practical task 6.1.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
+list1 = []
```
Comment (kolyasalubov) [inline YanZhylavy/HW6/Practical task 6.1.py:1]: good job

File: YanZhylavy/HW6/Practical task 6.2.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
+login = input('Please, enter your login ( Here is an advise: try "First" \U0001f609): ')
```
Comment (kolyasalubov) [inline YanZhylavy/HW6/Practical task 6.2.py:1]: good job

File: YanZhylavy/HW5/Practical task 5.3.py [hunk 11]
```diff
@@ -0,0 +1,6 @@
+n = int(input('Enter a number to calculate factorial:'))
```
Comment (kolyasalubov) [inline YanZhylavy/HW5/Practical task 5.3.py:1]: good job

File: YanZhylavy/HW5/Practical task 5.2.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+elements = int(input('''Please, tell how many elements
```
Comment (kolyasalubov) [inline YanZhylavy/HW5/Practical task 5.2.py]: good job, 
please look at this
n = int(input('enter some number from fibonacci list= '))

fibonacci_numbers = [0, 1]

for i in range(2,n):
        fibonacci_numbers.append(fibonacci_numbers[i-1]+fibonacci_numbers[i-2])
print(f'there are some numbers from fibanacci till that you entered {n},\n {fibonacci_numbers}')

Comment (YanZhylavy) [reply YanZhylavy/HW5/Practical task 5.2.py]: Thank you, fixed!

File: YanZhylavy/HW5/Practical task 5.1.py (hunk 8 above)
Comment (YanZhylavy) [reply YanZhylavy/HW5/Practical task 5.1.py]: Thank you, fixed!

File: YanZhylavy/HW5/Practical task 5.2.py [hunk 13]
```diff
@@ -0,0 +1,13 @@
+try:
```
Comment (kolyasalubov) [inline YanZhylavy/HW5/Practical task 5.2.py:1]: good job

File: YanZhylavy/HW7/Practical task 7.1.py [hunk 14]
```diff
@@ -0,0 +1,40 @@
+def largest_num(num1:int,num2:int) ->int:
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.1.py:1]: good job

File: YanZhylavy/HW7/Practical task 7.2.py [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+def circle_area(rad:float)->float:
+  """
+  Function calculates area of a circle
+  using the radius entered by the user 
+  """
+  PI = 3.14  
+  return PI*rad**2
+    
+
+greetings = input("""Hello! With this programm you can
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.2.py:24]: greetings = input("""Hello! With this programm you can
calculate area of rectangle, triangle or circle. Please,select
a figure: """).lower()

File: YanZhylavy/HW7/Practical task 7.2.py [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+calculate area of rectangle, triangle or circle. Please,select
+a figure: """)
+
+if greetings == "Triangle" or greetings == "triangle":
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.2.py:28]: if greetings == "triangle":

File: YanZhylavy/HW7/Practical task 7.2.py [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+    base = input("Enter base:")
+    height = input("Enter height:")
+    print("The answer is: ", triangle_area(float(base),float(height)))
+
+elif greetings == "Rectangle" or greetings == "rectangle":
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.2.py:33]: elif  greetings == "rectangle":

File: YanZhylavy/HW7/Practical task 7.2.py [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    len = input("Enter lenght: ")
+    wid = input("Enter widght: ")
+    print("The answer is: ", rectanle_area(float(len),float(wid)))
+
+elif greetings == "Circle" or greetings == "circle":
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.2.py:38]: elif greetings == "circle":

File: YanZhylavy/HW7/Practical task 7.3.py [hunk 19]
```diff
@@ -0,0 +1,10 @@
+def characters(word:str)->dict:
```
Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.3.py:1]: good job

Comment (kolyasalubov) [inline YanZhylavy/HW7/Practical task 7.3.py:1]: I recommended changing the function name:
maybe like this "character_count()"

File: YanZhylavy/HW 8/definitions.py [hunk 20]
```diff
@@ -0,0 +1,21 @@
+import math
```
Comment (kolyasalubov) [inline YanZhylavy/HW 8/definitions.py:1]: good job

File: YanZhylavy/HW 8/Practical task 8.3.py [hunk 21]
```diff
@@ -0,0 +1,19 @@
+import definitions as defi
```
Comment (kolyasalubov) [inline YanZhylavy/HW 8/Practical task 8.3.py:1]: good job

File: YanZhylavy/HW 8/Practical task 8.2.py [hunk 22]
```diff
@@ -0,0 +1,34 @@
+password = input('Enter password: ')
```
Comment (kolyasalubov) [inline YanZhylavy/HW 8/Practical task 8.2.py:1]: optimized version of your implementation

password = input('Enter password: ')
criteria = {'digits': False, 'uppers': False, 'lowers': False, 'sym': False, 'length': False}
symbols = ['#', '@', '$']

for i in password:
    if i.isdigit():
        criteria['digits'] = True
    elif i.isupper():
        criteria['uppers'] = True
    elif i.islower():
        criteria['lowers'] = True
    elif i in symbols:
        criteria['sym'] = True

if 6 <= len(password) <= 16:
    criteria['length'] = True

if all(criteria.values()):
    print('Password is valid')
else:
    print('Password is invalid')

File: YanZhylavy/HW 8/Practical task 8.1/utils/__init__.py [hunk 23]
```diff
@@ -0,0 +1,3 @@
+from .logger import*
```
Comment (kolyasalubov) [inline YanZhylavy/HW 8/Practical task 8.1/utils/__init__.py:1]: good job

**Pull request merged**